### PR TITLE
Fix friends and groups page features disappearing after navigation

### LINKED ISSUE #375: Detect webnavigation changes
So, Steam has some web pages that can load dynamically. This is problematic, as this will remove any augmentation done by the extension. It has to be redone.
To also change the URL, they use the `history.pushstate` function. Sadly, modern web browsers have no event for this. Chrome extensions however, can make use of `webNavigation` to detect this event, by attaching a listener to `chrome.webNavigation.onHistoryStateUpdated`. It does require you to add additional `webNavigation` permission.
I'd add this feature myself, but I am not entirely sure how you would this to be implemented. Also, this listener has to be done in a background script. So for other scripts to communicate we probably have to use some messages system.

Here is an example:
1. Go to https://steamcommunity.com/my/friends
2. Observe the friend sorting augmentation that was added by Augmented Steam.
3. Now load another page by clicking e.g. "Add a friend"
4. Now, go back to "Your friends"
5. Observe how the friend sorting augmentation is now missing

### PR DESCRIPTION
**Pull request by candela97**
Fixes #375 for friends and groups page.

I took the approach of merging `CFriends` and `CGroups` into a single context `CFriendsAndGroups`, turning the features into callbacks and listening to ajax success events so they can be triggered after sub-frame navigation.

### CODE REVIEW

File: src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            FInviteFriendsToGroup,
+            FGroupsManageButton,
+            FGroupsSort,
+        ]);
+
+        this.myProfile = CommunityUtils.currentUserIsOwner();
+
+        Page.runInPageContext(() => {
+            window.SteamFacade.jq(document).ajaxSuccess((event, xhr, settings) => {
+                if (settings.url.endsWith("friends?ajax=1") || settings.url.endsWith("groups/?ajax=1")) {
```
Comment (candela97) [inline src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js:24]: these checks can probably be removed...the features check if the correct sub-frame is loaded already

File: src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            FInviteFriendsToGroup,
+            FGroupsManageButton,
+            FGroupsSort,
+        ]);
+
+        this.myProfile = CommunityUtils.currentUserIsOwner();
+
+        Page.runInPageContext(() => {
+            window.SteamFacade.jq(document).ajaxSuccess((event, xhr, settings) => {
+                if (/\/(friends|groups)\/?\?ajax=1$/.test(settings.url)) {
```
Comment (Revadike) [inline src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js:24]: Maybe parse the URL instead? Using regex can break easily and may not cover everything.

File: src/js/Content/Features/Community/FriendsAndGroups/FGroupsManageButton.js [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
 
-    apply() {
+    setup() {
+        this.callback();
+    }
+
+    callback() {
+        if (!document.getElementById("groups_list")) { return; }
+
+        this._groups = Array.from(document.querySelectorAll(".group_block"));
```
Comment (Revadike) [inline src/js/Content/Features/Community/FriendsAndGroups/FGroupsManageButton.js:18]: I don't think `Array.from` is needed in this case.

File: src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js (hunk 2 above)
Comment (candela97) [reply src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js:24]: I wasn't sure either...I can revert the last commit if that's better.

Comment (Revadike) [reply src/js/Content/Features/Community/FriendsAndGroups/CFriendsAndGroups.js:24]:
```suggestion
const url = new URL(settings.url);
if (url.searchParams.has("ajax") && (url.pathname.split("/")[3] === "friends" || url.pathname.split("/")[3] === "groups")) {
```

**Pull request merged**